src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -708,6 +704,63 @@ private async Task<IEnumerable<DiagnosticApiResponse>> ListDetectorsInternal(Run
                 }
             }
 
+            if (tableCacheService.IsStorageAsSourceEnabled())
+            {
+                var allDetectorsFromStorage = await tableCacheService.GetEntityListByType<TResource>(context);
+                if (searchResults != null)
+                {
+                    // Return those detectors that have positive search score and present in searchResult.
+                    List<string> potentialDetectors = searchResults.Results.Where(s => s.Score > 0).Select(x => x.Detector).ToList();
+
+                    // Assign the score to detector if it exists in search results, else default to 0
+                    allDetectorsFromStorage.ForEach(entity =>
+                    {
+                        var detectorWithScore = (searchResults != null) ? searchResults.Results.FirstOrDefault(x => x.Detector == entity.RowKey) : null;
+                        entity.Score = detectorWithScore != null ? detectorWithScore.Score : 0;
+                    });
+
+                    // Filter only postive score detectors.
+                    allDetectorsFromStorage = allDetectorsFromStorage.Where(x => x.Score > 0).ToList();
+                }
```

Comment:
Should we add the logging on search results at the end of if block here - Just like we are doing below for sourcewatcher results.

var logMessage = new { InsightName = "SearchResultsPublic", InsightData = allDetectorsFromStorage.Select(p => new { Id = p.RowKey, Score = p.Score }) };
                DiagnosticsETWProvider.Instance.LogInternalAPIInsights(context.OperationContext.RequestId, JsonConvert.SerializeObject(logMessage));

Reply:
Sure, I missed it. I will add it